controllers/extensionRequests.js

```diff
@@ -132,22 +225,45 @@ const getExtensionRequest = async (req, res) => {
  * @param res {Object} - Express response object
  */
 const getSelfExtensionRequests = async (req, res) => {
-  try {
-    const { id: userId } = req.userData;
-    const { taskId, status } = req.query;
-
-    if (userId) {
-      const allExtensionRequests = await extensionRequestsQuery.fetchExtensionRequests({
-        status,
-        taskId,
-        assignee: userId,
-      });
-      return res.json({ message: "Extension Requests returned successfully!", allExtensionRequests });
+  // Inside dev flag it just shows
+  const dev = req.query.dev === "true";
+  if (dev) {
+    try {
+      const { id: userId } = req.userData;
+      const { taskId } = req.query;
+
+      if (userId) {
+        let allExtensionRequests = await extensionRequestsQuery.fetchLatestExtensionRequest({
+          taskId,
+        });
+
+        if (allExtensionRequests.length && allExtensionRequests[0].userId !== userId) {
+          allExtensionRequests = [];
+        }
```

Comment:
Same case as mentioned [here](https://github.com/Real-Dev-Squad/website-backend/pull/1563/files#r1345925185). 

In this case wouldn't the API return an empty array? even though there are older extension request.